cpp/models/ide_secir/model.cpp

```diff
@@ -242,213 +315,283 @@ void Model::compute_flow(Eigen::Index idx_InfectionTransitions, Eigen::Index idx
 
     Hence calc_time_index goes until std::ceil(support_max/dt) since for std::ceil(support_max/dt)+1 all terms are already zero. 
     This needs to be adjusted if we are changing the finite difference scheme */
-
     Eigen::Index calc_time_index = (Eigen::Index)std::ceil(
-        parameters.get<TransitionDistributions>()[idx_InfectionTransitions].get_support_max(dt, m_tol) / dt);
+        parameters.get<TransitionDistributions>()[group][idx_InfectionTransitions].get_support_max(dt, m_tol) / dt);
 
+    int Transition_idx = get_transition_flat_index(idx_InfectionTransitions, static_cast<Eigen::Index>((size_t)group));
```

Comment:
```suggestion
    int transition_idx = get_transition_flat_index(idx_InfectionTransitions, static_cast<Eigen::Index>((size_t)group));
```
For consistency, most variables start with a small letter